### PR TITLE
feat: Also release as a Home Assistant add-on

### PR DESCRIPTION
**Pull request by ronaldheft**
This PR adds the minimal necessary files to release as a Home Assistant add-on. This uses the existing Docker image as-is and simply provides a convenience to setup on Home Assistant OS.

When installed as an add-on, the configuration will still be done via the config.yml file. In the future the add-on could take advantage of [add-ons' options schema](https://developers.home-assistant.io/docs/add-ons/configuration#options--schema) to generate the configuration or even automatically configure MQTT.

I've left this PR in draft-mode for the moment as I'm not sure how to handle the version strings in `build.yml` and `config.yml`. The release-please GitHub Action will not automatically update the version strings on the PR it creates. I see three options:

1. Manually update the version strings after each release
2. Take the files, create a separate repo (the add-on can be in its own repo), and manually update those files
3. Write custom code or find an alternative to handle updating the version strings as the GitHub release action

Thoughts @zachowj?

### CODE REVIEW

File: xfinity-data-usage/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,2 @@
+ARG BUILD_FROM
+FROM $BUILD_FROM
```
Comment (ronaldheft) [inline xfinity-data-usage/Dockerfile:2]: A Docker container is build using existing published image. A Dockerfile is required in the add-on directory, so this file cannot be omitted.

File: xfinity-data-usage/config.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+startup: system
+boot: auto
+options: {}
+schema: {}
+ports:
+  7878/tcp: null
+ports_description:
+  7878/tcp: REST sensor
+map:
+  - 'config:rw'
```
Comment (ronaldheft) [inline xfinity-data-usage/config.yml:22]: Allow read-write of the `/config` folder.

File: xfinity-data-usage/config.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  - armv7
+  - aarch64
+  - amd64
+  - i386
+startup: system
+boot: auto
+options: {}
+schema: {}
+ports:
+  7878/tcp: null
```
Comment (ronaldheft) [inline xfinity-data-usage/config.yml:18]: Allow egress for the REST sensor.

File: xfinity-data-usage/config.yml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+description: >-
+  Fetch Xfinity data usage and serve it via an HTTP endpoint, publish it to MQTT
+  or post it to an URL.
+arch:
+  - armhf
+  - armv7
+  - aarch64
+  - amd64
+  - i386
+startup: system
```
Comment (ronaldheft) [inline xfinity-data-usage/config.yml:13]: Run independent of Home Assistant.

File: xfinity-data-usage/config.yml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+ports:
+  7878/tcp: null
+ports_description:
+  7878/tcp: REST sensor
+map:
+  - 'config:rw'
+environment:
+  TZ: US/Eastern
+  LOGGING_LEVEL: debug
+  CONFIG_FOLDER: /config/xfinity-data-usage
```
Comment (ronaldheft) [inline xfinity-data-usage/config.yml:26]: Override the config folder location. Line 22 maps Home Assistant's `/config` directory into the container.

**Comment by zachowj** [comment]
I know close to zero about the development of a HA add-on as I don't run HA with supervisor.

As far as what path should be taken to release the standard docker and the add-on images, I don't have an opinion either way. If you want to add it to this repo in whatever fashion I'm all for it. I can add you as a collaborator to make things easier.

I only use release-please because it's what I use on my other open-source projects and it makes doing releases simple.  But if there's a better way to be able to release both of the images I don't mind changing to it.

For the versions in `build.yml`. Do you have to pin a version, or can you use `latest`?

**Comment by zachowj** [comment]
Also currently, this package is only using buildx for the following platforms. So I don't know if that matters for the `build.yml` file.

 DOCKER_PLATFORMS=linux/amd64,linux/arm64,linux/arm/v7

**Comment by zachowj** [comment]
> I'm not sure how to handle the version strings in `build.yml` and `config.yml`

If they only need to be set as build time a simple GH action could be used to replace a placeholder with actual values.

https://github.com/jacobtomlinson/gha-find-replace

```
build_from:
  armhf: zachowj/xfinity-data-usage:%%version%%
  armv7: zachowj/xfinity-data-usage:%%version%%
  aarch64: zachowj/xfinity-data-usage:%%version%%
  amd64: zachowj/xfinity-data-usage:%%version%%
  i386: zachowj/xfinity-data-usage:%%version%%
```

**Comment by ronaldheft** [comment]
@zachowj Sorry, for just getting back now; been crazy busy this week!

> For the versions in build.yml. Do you have to pin a version, or can you use latest?

> If they only need to be set as build time a simple GH action could be used to replace a placeholder with actual values.
>
> https://github.com/jacobtomlinson/gha-find-replace

For the `build.yml`, using placeholder variable is definitely possible, but it would need to be done in addition to publishing a pre-built image.

Currently, the way the add-on is configured, Home Assistant builds a Docker container locally. Since the `Dockerfile` is only inheriting the existing image, it's not building much, but its still reading the `build.yml` from the repo directly.

For `config.yml`, it has to be the actual version tag. Home Assistant add-ons will always read that from the Git repo (and tagged releases cannot be used :-( ).

> As far as what path should be taken to release the standard docker and the add-on images, I don't have an opinion either way. If you want to add it to this repo in whatever fashion I'm all for it. I can add you as a collaborator to make things easier.
> 
> I only use release-please because it's what I use on my other open-source projects and it makes doing releases simple. But if there's a better way to be able to release both of the images I don't mind changing to it.

I'm thinking it will make the most sense to create the add-on in a different repo, that way an entirely different release process can be used without conflicting with the current release GitHub Action.

I'll create a new repo, or if you'd like to keep the add-on under your GitHub handle, you could create `zachowj/xfinity-data-usage-addon` and add me on as a collaborator there.

Either way, when I have more time this weekend, I'll move it to it's own repo, script up a release Action that handles the HA add-on tagging, and then update this PR only include the documentation of where to download the add-on.

> Also currently, this package is only using buildx for the following platforms. So I don't know if that matters for the build.yml file.
> 
> DOCKER_PLATFORMS=linux/amd64,linux/arm64,linux/arm/v7

Great call out. I need to adjust to build only with supported platforms.

**Comment by zachowj** [comment]
> I'm thinking it will make the most sense to create the add-on in a different repo, that way an entirely different release process can be used without conflicting with the current release GitHub Action.

Sounds good. I see no reason why it needs to be under my handle.

> Great call out. I need to adjust to build only with supported platforms.

I don't think there's a reason it can't be built for the other platforms. I didn't do it originally because I had no way to test them and I guess I still don't.

**Comment by jlambert121** [comment]
@ronaldheft did you ever decide to create this as an add-on?  If so, do you mind linking here?

**Comment by ronaldheft** [comment]
@jlambert121 I do have it available as an add-on. It requires me to manually update the repository whenever a new release is made, but I just updated the add-on for the recently released 0.17.8 version.

https://github.com/ronaldheft/hass-addons

**Comment by Ali229** [comment]
> @jlambert121 I do have it available as an add-on. It requires me to manually update the repository whenever a new release is made, but I just updated the add-on for the recently released 0.17.8 version.
> 
> https://github.com/ronaldheft/hass-addons

Can you please explain how to use this? I really need the xfinity usage tracked but can't make sense of step by step on how to set this up on my home assistant.

**Pull request closed without merging**